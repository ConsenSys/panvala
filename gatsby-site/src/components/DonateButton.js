import React from 'react';

function DonateButton({ handleClick, text, disabled }) {
  return (
    <div>
      <input
        type="submit"
        name="submit"
        onClick={handleClick}
        className="f6 link dim bn br-pill pv3 ph4 white bg-teal fw7 mt4"
        disabled={disabled}
        value={text ? text : 'Donate'}
      />
    </div>
  );
}

export default DonateButton;
