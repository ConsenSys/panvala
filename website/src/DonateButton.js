'use strict';

function DonateButton({ handleClick }) {
  return (
    <div>
      <input
        type="submit"
        name="submit"
        onClick={handleClick}
        className="f6 link dim bn br-pill pv3 ph4 white bg-teal fw7 mt4"
        value="Donate"
      />
    </div>
  );
}
