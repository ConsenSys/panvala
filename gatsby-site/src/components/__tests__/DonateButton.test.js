import React from 'react';
import DonateButton from '../DonateButton';
import { create } from 'react-test-renderer';

describe('DonateButton', () => {
  const clickHandler = jest.fn();

  test('should render component', () => {
    // Test render
    const component = create(<DonateButton handleClick={clickHandler} />);
    const instance = component.root;

    // Get input (submit button) element
    const button = instance.findByType('input')

    // Mock: click button
    button.props.onClick();
    expect(clickHandler).toHaveBeenCalled();

    // Snapshot
    expect(component.toJSON()).toMatchSnapshot();
  });
});
