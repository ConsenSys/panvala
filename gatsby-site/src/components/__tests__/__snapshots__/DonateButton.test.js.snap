// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DonateButton should render component 1`] = `
<div>
  <input
    className="f6 link dim bn br-pill pv3 ph4 white bg-teal fw7 mt4"
    name="submit"
    onClick={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    type="submit"
    value="Donate"
  />
</div>
`;
